### PR TITLE
Implemented ActionQueue hud display and clear-one/clear-all buttons.

### PR DESCRIPTION
**Pull request by uwadmin12**
Best to merge #22 before this

Demo: https://imgur.com/a/VELJ1y6

### CODE REVIEW

File: src/game/gui/hud.cpp [hunk 1]
```diff
@@ -201,6 +271,19 @@ void HUD::onClick(const string &control) {
         _game->openInGameMenu(InGameMenu::Tab::Map);
     } else if (control == "BTN_OPT") {
         _game->openInGameMenu(InGameMenu::Tab::Options);
+    } else if (control == "BTN_CLEARALL") {
+        _game->party().leader()->actionQueue().clear();
+    } else if (control == "BTN_CLEARONE" || control == "BTN_CLEARONE2") {
+        auto &actionQueue = _game->party().leader()->actionQueue();
+
+        for (auto it=actionQueue.begin(); it!=actionQueue.end(); ++it) {
```
Comment (seedhartha) [inline src/game/gui/hud.cpp:279]: Would this compile?

```cpp
for (auto &action : actionQueue)
```

File: src/game/gui/hud.cpp [hunk 2]
```diff
@@ -184,6 +192,68 @@ void HUD::render() const {
...
+    hideControl("LBL_QUEUE0");
+    hideControl("LBL_QUEUE1");
+    hideControl("LBL_QUEUE2");
+    hideControl("LBL_QUEUE3");
+}
+
+void HUD::drawActionQueueItems() const {
+    auto &actionQueue = _game->party().leader()->actionQueue();
+
+    // clear the drawn queue items first
```
Comment (seedhartha) [inline src/game/gui/hud.cpp:235]: Maybe add actions first, and clear the rest of the action queue in a loop from `i` until 4?

**Review by seedhartha — CHANGES_REQUESTED** [review]

File: src/game/gui/hud.cpp (hunk 2 above)
Comment (seedhartha) [reply src/game/gui/hud.cpp:235]: Actually, these two loops are asking to be combined.

```cpp
for (int i = 0; i < 4; ++i) {
    string fill;
    if (it != actionQueue.end()) {
        if ((*it)->type() = ActionType::AttackObject) {
           fill = "i_attack";     
        }
        ++it;
    }
    Control &item = getControl("LBL_QUEUE" + to_string(i));
    item.setBorderFill(fill);
}
```

File: src/game/gui/hud.cpp (hunk 1 above)
Comment (uwadmin12) [reply src/game/gui/hud.cpp:279]: It would, actually. 👀

File: src/game/gui/hud.cpp (hunk 2 above)
Comment (uwadmin12) [reply src/game/gui/hud.cpp:235]: I wanted to account for non-displayable actions in the queue as well (which may be included in the future?) So what about this:
```c++
    auto it = actionQueue.begin();
    for (int i = 0; i < 4; ++i) {
        string fill;

        while (it != actionQueue.end() && (*it)->type() != ActionType::AttackObject)
            ++it;
        if (it != actionQueue.end()) {
            fill = "i_attack";
        }
        Control& item = getControl("LBL_QUEUE" + to_string(i));
        item.setBorderFill(fill);
    }
```

**Comment by seedhartha** [comment]
Merged, thanks!

**Pull request merged**